Modules/Devices/test/Devices.Domain.Tests/Identities/ApproveDeletionProcessTests.cs

```diff
@@ -30,29 +31,47 @@ public void Approve_deletion_process_waiting_for_approval()
         AssertAuditLogEntryWasCreated(deletionProcess);
     }
 
+    [Fact]
+    public void Throws_when_device_not_owned_by_identity()
+    {
+        // Arrange
+        SystemTime.Set(DateTime.Parse("2020-01-01"));
+        var identity = CreateIdentityWithDeletionProcessWaitingForApproval();
+
+        // Act
+        var acting = () => identity.ApproveDeletionProcess(identity.DeletionProcesses[0].Id, DeviceId.Parse("DVC"));
+
+        // Assert
+        acting.Should().Throw<DomainException>().Which.Code.Should().Be("error.platform.recordNotFound");
+    }
+
     [Fact]
     public void Throws_when_deletion_process_does_not_exist()
     {
         // Arrange
         var identity = CreateIdentity();
-        var identityDeletionProcessId = IdentityDeletionProcessId.Create("IDP00000000000000001").Value;
+        identity.Devices.Add(new Device(identity));
+        var deviceId = identity.Devices[0].Id;
+        var deletionProcessId = IdentityDeletionProcessId.Create("IDP00000000000000001").Value;
 
         // Act
-        var acting = () => identity.ApproveDeletionProcess(identityDeletionProcessId, DeviceId.Parse("DVC"));
+        var acting = () => identity.ApproveDeletionProcess(deletionProcessId, deviceId);
 
         // Assert
-        acting.Should().Throw<DomainException>().Which.Code.Should().Be("error.platform.recordNotFound");
+        acting.Should().Throw<DomainException>().Which.Message.Should().Contain("IdentityDeletionProcess");
```

Comment:
Now you removed the check for the error code and only check for the error message. 😞

Reply:
I understood there should only be one. Your comment was:
> Assert that the message contains `IdentityDeletionProcess`

I agree it can be understood as both "replace the existing with suggested" and "in addition to existing add the suggested", I simply picked the wrong one.